### PR TITLE
[RFC] Add sample scrapigen metadata

### PR DESCRIPTION
**Pull request by subash-suresh**
- This PR adds a sample scrapigen metadata that can used to understand the metadata that is expected from scrapigen generator. This data can also be used as a reference to write code generation logic for generating nidaqmx python files.
- The docuementation for the metadata have also been added as part of this PR.
- The `enums.py` is the same file that is being generated currently by scrapigen. The additional information that would be added on to the `enums.py` would be,
    - metadata on the [daqmx error codes](https://github.com/ni/nidaqmx-python/blob/master/nidaqmx/error_codes.py) (which is enum ) 
- The other two files namely the functions.py and attributes.py have just a few fields to just demonstrate the metadata that is required to be added. This is just for showing the metadata requirement and the final version of the metadata would contain all the possible functions and attributes.

### CODE REVIEW

**Comment by zhindes** [comment]
No concerns adding this information to the repo. One request from me is to please add a README to `generator/nidaqmx_python_generator/metadata` (or one level up in `generator/nidaqmx_python_generator`) that explains where this stuff came from. I would also love it if you copied the information from your [metadata details spreadsheet](https://nio365.sharepoint.com/:x:/s/ETWMFGlobalCollaboration/ETsIvMKwvH9NnZCFkZklLIoBektqElJ6Jh39S-jg0OUKkg?e=pAq3jg) in there, too.

**Comment by bkeryan** [comment]
> No concerns adding this information to the repo. One request from me is to please add a README to `generator/nidaqmx_python_generator/metadata` (or one level up in `generator/nidaqmx_python_generator`) that explains where this stuff came from. I would also love it if you copied the information from your [metadata details spreadsheet](https://nio365.sharepoint.com/:x:/s/ETWMFGlobalCollaboration/ETsIvMKwvH9NnZCFkZklLIoBektqElJ6Jh39S-jg0OUKkg?e=pAq3jg) in there, too.

Make it a README.md so you can use Markdown.

https://dev.azure.com/ni/DevCentral/_git/ni-central?path=/src/hapigen/hapigen_documentation/base&_a=contents&version=GBmain might give you some ideas regarding documentation formatting.

File: generator/nidaqmx_python_generator/metadata/attributes.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+attributes = {
+    'Channel': {
+         6145: {
+            'access': 'read-write',
+            'name': 'AI_ATTEN',
+            'resettable': True,
+            'type': 'float64',
+            'ctypesDataType': 'ctypes.c_double',
```
Comment (bkeryan) [inline generator/nidaqmx_python_generator/metadata/attributes.py:8]: Should scrapigen keys be in camelCase or snake_case? I'm not sure if there is a precedent.

File: generator/nidaqmx_python_generator/metadata/attributes.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            'callingConvention':'StdCall',
+            'cFunctionName':'AIAtten',
+            'handleParameters':
+            {
+                'taskHandle':{
+                    "accessor": "self._handle", 
+                    "ctypesDataType": "lib_importer.task_handle", 
+                    "cviName": "taskHandle"
+                },
+                'ChannelHandle':{
```
Comment (bkeryan) [inline generator/nidaqmx_python_generator/metadata/attributes.py:23]: This is a name, not a handle.

Please decide whether these `handleParameters` keys are in camelCase, PascalCase, or snake_case.

**Comment by zhindes** [comment]
How did you change the naming of the keys in the metadata? Are you modifying the scrapigen generator?

File: generator/nidaqmx_python_generator/metadata/doc/functions.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+- `'description'`
+    - The description of the function.
+    - This is used to define the doc string of the function.
+
+- `'calling_convention'`
+    - The calling convention to be followed when using the c functions.
+    - Possible values
+        - 'StdCall'
+        - 'Cdecl'
+- `'handle_parameter'`(Optional)
```
Comment (bkeryan) [inline generator/nidaqmx_python_generator/metadata/doc/functions.md:53]: When writing English, always leave a space between parentheses and the surrounding words.

File: generator/nidaqmx_python_generator/metadata/doc/functions.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        'c_function_name' : 'CreateAIAccel4WireDCVoltageChan', 
+        'calling_convention':'StdCall',
+        'description': ' Creates channel(s) that use a Wheatstone bridge to measure force or load. Use this instance with sensors whose...',
+        'isFactory': False,
+        'python_class_name':'AIChannelCollection',
+        'handle_parameter':
+        {
+            "accessor": "self._handle", 
+            "ctypes_data_type": "lib_importer.task_handle", 
+            "cviName": "taskHandle"
```
Comment (bkeryan) [inline generator/nidaqmx_python_generator/metadata/doc/functions.md:16]: `isFactory` and `cviName` are still camelCase.

File: generator/nidaqmx_python_generator/metadata/attributes.py (hunk 1 above)
Comment (bkeryan) [reply generator/nidaqmx_python_generator/metadata/attributes.py]: Was there a precedent? (i.e. did the existing scrapigen output have any camelCase or snake_case names?)

File: generator/nidaqmx_python_generator/metadata/attributes.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                    "accessor": "self._name", 
+                    "ctypes_data_type": "ctypes.c_char_p", 
+                    "cviName": "channel"
+                }
+            },
+            'is_object': True,
+            'object_type': 'Channel',
+            'object_has_factory': True,
+            'objectConstructorParams':
```
Comment (bkeryan) [inline generator/nidaqmx_python_generator/metadata/attributes.py:119]: `objectConstructorParams` is still camelCase.

Comment (subash-suresh) [reply generator/nidaqmx_python_generator/metadata/attributes.py:119]: @bkeryan I saw some of the other multi worded metadata in functions.py for daqmx which were of snake case (these were grpc related metadata), hence i moved all data to snake case

**Comment by zhindes** [comment]
> 

Ahh ok. That makes sense!

File: generator/nidaqmx_python_generator/metadata/attributes.py (hunk 5 above)
Comment (bkeryan) [reply generator/nidaqmx_python_generator/metadata/attributes.py:119]: ok, I see: https://github.com/ni/grpc-device/blob/main/source/codegen/metadata/nidaqmx/functions.py#L5625

```
    'CreateTask': {
        'init_method': True,
        'parameters': [
            {
                'direction': 'in',
                'is_session_name': True,
                'name': 'sessionName',
                'type': 'const char[]'
            },
            {
                'direction': 'out',
                'name': 'task',
                'type': 'TaskHandle'
            },
            {
                'cppName': 'initializationBehavior',
                'direction': 'in',
                'grpc_type': 'nidevice_grpc.SessionInitializationBehavior',
                'name': 'initializationBehavior',
                'proto_only': True,
                'type': 'int32'
            },
            {
                'cppName': 'newSessionInitialized',
                'direction': 'out',
                'grpc_type': 'bool',
                'name': 'newSessionInitialized',
                'proto_only': True,
                'type': 'bool'
            }
        ],
        'returns': 'int32'
    },
```

File: generator/nidaqmx_python_generator/metadata/attributes.py (hunk 1 above)
Comment (subash-suresh) [reply generator/nidaqmx_python_generator/metadata/attributes.py]: as mentioned, I noticed some other [existing keys related to grpc proto generation](https://github.com/ni/grpc-device-scrapigen/blob/30746b2ee2a166ee3dfcd8f8a77550a5e85c8a9f/out/daq/export/functions.py#L5677) being snake case, therefore I switched my implementation to snake case as well.

**Review by dixonjoel — APPROVED** [review]

File: generator/nidaqmx_python_generator/metadata/attributes.py (hunk 2 above)
Comment (subash-suresh) [reply generator/nidaqmx_python_generator/metadata/attributes.py]: updated to use names

File: generator/nidaqmx_python_generator/metadata/doc/functions.md (hunk 4 above)
Comment (subash-suresh) [reply generator/nidaqmx_python_generator/metadata/doc/functions.md]: changed them to snake case

File: generator/nidaqmx_python_generator/metadata/attributes.py (hunk 5 above)
Comment (subash-suresh) [reply generator/nidaqmx_python_generator/metadata/attributes.py]: So based on this we could stick with snake case right?

**Pull request closed without merging**